ocaml/idl/datamodel_vtpm.ml

```diff
@@ -39,24 +67,30 @@ let t =
       [
         (Published, rel_rio, "Added VTPM stub")
       ; (Extended, rel_next, "Added ability to manipulate contents")
-      ; (Extended, rel_next, "Added VTPM profiles")
-      ; (Changed, rel_next, "Removed backend field")
+      ; (Extended, rel_next, "Added VTPM unique and protected properties")
+      ; (Extended, rel_next, "Added Persistence backed")
       ]
-    ~gen_constructor_destructor:true ~name:_vtpm ~descr:"A virtual TPM device"
-    ~gen_events:false ~doccomments:[]
+    ~gen_constructor_destructor:false ~name:_vtpm ~descr:"A virtual TPM device"
+    ~gen_events:true ~doccomments:[]
     ~messages_default_allowed_roles:_R_POOL_ADMIN
     ~contents:
       [
         uid _vtpm
       ; field ~qualifier:StaticRO ~ty:(Ref _vm) "VM"
           "The virtual machine the TPM is attached to"
-      ; field ~qualifier:DynamicRO
-          ~ty:(Map (String, String))
-          ~lifecycle:[(Published, rel_next, "Added VTPM profiles")]
-          "profile" "The security properties that define how the TPM is handled"
+      ; field ~qualifier:StaticRO ~ty:(Ref _vm) "backend"
```

Comment:
Even though you are explicit defining the `create` function, please keep the qualifier's in sync with its arguments, just to avoid confusion. So `VM` and `unique` should be StaticRO, and all others DynamicRO.